DeepSpeech.py

```diff
@@ -0,0 +1,1164 @@
+
+
+# Sample limits
+
+# The maximum amount of samples taken from (the beginning of) the train set - 0 meaning no limit
+limit_train = int(os.environ.get('ds_limit_train', 0))
+
+# The maximum amount of samples taken from (the beginning of) the validation set - 0 meaning no limit
+limit_dev   = int(os.environ.get('ds_limit_dev',   0))
+
+# The maximum amount of samples taken from (the beginning of) the test set - 0 meaning no limit
+limit_test  = int(os.environ.get('ds_limit_test',  0))
+
+
+# Step widths
+
+# The number of epochs we cycle through before displaying progress
+display_step = int(os.environ.get('ds_display_step', 1))
+
+# The number of epochs we cycle through before checkpointing the model
+checkpoint_step = int(os.environ.get('ds_checkpoint_step', 5))
+
+# The number of epochs we cycle through before validating the model
+validation_step = int(os.environ.get('ds_validation_step', 0))
+
+
+# Checkpointing
+
+# The directory in which checkpoints are stored
+checkpoint_dir = os.environ.get('ds_checkpoint_dir', xdg.save_data_path('deepspeech'))
+
+# Whether to resume from checkpoints when training
+restore_checkpoint = bool(int(os.environ.get('ds_restore_checkpoint', 0)))
+
+# The directory in which exported models are stored
+export_dir = os.environ.get('ds_export_dir', None)
+
+# The version number of the exported model
+export_version = 1
+
+# Whether to remove old exported models
+remove_export = bool(int(os.environ.get('ds_remove_export', 0)))
+
+
+# Reporting
+
+# The number of phrases to print out during a WER report
+report_count = int(os.environ.get('ds_report_count', 10))
+
+# Whether to log device placement of the operators to the console
+log_device_placement = bool(int(os.environ.get('ds_log_device_placement', 0)))
+
+# Whether to log gradients and variables summaries to TensorBoard during training.
+# This incurs a performance hit, so should generally only be enabled for debugging.
+log_variables = bool(len(os.environ.get('ds_log_variables', '')))
+
+
+# Initialization
+
+# The default random seed that is used to initialize variables. Ensures reproducibility.
+random_seed = int(os.environ.get('ds_random_seed', 4567)) # To be adjusted in case of bad luck
+
+# The default standard deviation to use when initialising weights and biases
+default_stddev = float(os.environ.get('ds_default_stddev', 0.046875))
+
+# Individual standard deviations to use when initialising particular weights and biases
+for var in ['b1', 'h1', 'b2', 'h2', 'b3', 'h3', 'b5', 'h5', 'b6', 'h6']:
+    locals()['%s_stddev' % var] = float(os.environ.get('ds_%s_stddev' % var, default_stddev))
+
+# Session settings
+
+# Standard session configuration that'll be used for all new sessions.
+session_config = tf.ConfigProto(allow_soft_placement=True, log_device_placement=log_device_placement)
+
+
+# Geometric Constants
+# ===================
+
+# For an explanation of the meaning of the geometric constants, please refer to
+# doc/Geometry.md
```

Comment:
So I looked into Sphinx + GitHub. It's definitely doable, but it would require an additional build step for *any* documentation changes, not just the math stuff in the doc folder (as they exist in this PR).

IMO, the cleanest option would be setting up a GitHub hook to automatically rebuild the docs on every commit and publish on this repo's gh-pages branch. That way we could keep just updating comments here and docs would magically be available in https://mozilla.github.io/DeepSpeech

What do you think about this approach?